apps/dokploy/components/dashboard/settings/servers/gpu-support.tsx

```diff
@@ -0,0 +1,262 @@
+import { AlertBlock } from "@/components/shared/alert-block";
+import { DialogAction } from "@/components/shared/dialog-action";
+import { Button } from "@/components/ui/button";
+import {
+	Card,
+	CardContent,
+	CardDescription,
+	CardHeader,
+	CardTitle,
+} from "@/components/ui/card";
+import { api } from "@/utils/api";
+import { TRPCClientError } from "@trpc/client";
+import { CheckCircle2, Cpu, Loader2, XCircle } from "lucide-react";
+import { useState } from "react";
+import { toast } from "sonner";
+
+interface GPUSupportProps {
+	serverId?: string;
+}
+
+export function GPUSupport({ serverId }: GPUSupportProps) {
+	const [isLoading, setIsLoading] = useState(false);
+	const utils = api.useContext();
+
+	const { data: gpuStatus, isLoading: isChecking } =
+		api.settings.checkGPUStatus.useQuery(
+			{ serverId },
+			{
+				enabled: serverId !== undefined,
+				refetchInterval: 5000,
+			},
+		);
+
+	const setupGPU = api.settings.setupGPU.useMutation({
+		onMutate: () => {
+			setIsLoading(true);
+		},
+		onSuccess: async () => {
+			toast.success("GPU support enabled successfully");
+			setIsLoading(false);
+			await utils.settings.checkGPUStatus.invalidate({ serverId });
+		},
+		onError: (error) => {
+			if (error instanceof TRPCClientError) {
+				const errorMessage = error.message;
+				if (
+					errorMessage.includes(
+						"Permission denied. Please ensure proper sudo access.",
+					) ||
+					errorMessage.includes("sudo access required")
+				) {
+					toast.error(
+						"Administrator privileges required. Please enter your password when prompted.",
+					);
+				} else if (errorMessage.includes("Failed to configure GPU")) {
+					toast.error(
+						"GPU configuration failed. Please check system requirements.",
+					);
+				} else {
+					toast.error(errorMessage);
+				}
+			} else {
+				toast.error("Failed to enable GPU support. Please check server logs.");
+			}
+			setIsLoading(false);
```

Comment:
I'm pretty sure this can be abstracted from the backend side, I mean throw the error on the server and then when you trigger the `setupGPU.mutateAsync` catch the error and then show as err.message in the toaster and we can safely remove this code